File: patientregistration/src/main/java/com/bss/patientregistration/listener/TextInputLayoutFocusChangeListener.java
package com.bss.patientregistration.listener;

import android.view.View;

import com.google.android.material.textfield.TextInputEditText;
import com.google.android.material.textfield.TextInputLayout;

public class TextInputLayoutFocusChangeListener implements View.OnFocusChangeListener{
    TextInputLayout textInputLayout;
    String emptyMessage;
    public TextInputLayoutFocusChangeListener(TextInputLayout textInputLayout,String emptyMessage)
    {
        this.textInputLayout = textInputLayout;
        this.emptyMessage = emptyMessage;
    }
    @Override
    public void onFocusChange(View v, boolean hasFocus) {
        textInputLayout.setError(null);
        TextInputEditText textInputEditText = (TextInputEditText)v;
        String text = textInputEditText.getText().toString();
        if(null == text || text.isEmpty())
            textInputLayout.setError(emptyMessage);
    }
}

File: app/src/main/java/com/bss/uis/listner/TextInputLayoutFocusChangeListener.kt
package com.bss.uis.listner

import android.view.View
import com.google.android.material.textfield.TextInputEditText
import com.google.android.material.textfield.TextInputLayout

class TextInputLayoutFocusChangeListener(textInputLayout: TextInputLayout?,
                                         emptyMessage: String?) : View.OnFocusChangeListener {
    var textInputLayout: TextInputLayout? = textInputLayout
    var emptyMessage: String? = emptyMessage


    override fun onFocusChange(p0: View?, p1: Boolean) {
        textInputLayout!!.error = null
        val textInputEditText = p0 as TextInputEditText
        val text = textInputEditText.text.toString()
        if (text.isEmpty()) textInputLayout!!.error = emptyMessage
    }
}